algebird-core/src/main/scala/com/twitter/algebird/QTree.scala

```diff
@@ -108,32 +152,45 @@ class QTreeSemigroup[A](k: Int)(implicit val underlyingMonoid: Monoid[A]) extend
   }
 }
 
-case class QTree[A](
+case class QTree[@specialized(Int, Long, Float, Double, Unit) A] private[algebird] (
```

Comment:
I'm pretty sure the args here actually remain public, mima only reports the loss of some variants of this() constructors that I saw -- I'll run this through our internal stuff which should his tests before a merge